src/main/java/racingcar/domain/Cars.java

```diff
@@ -1,56 +1,54 @@
 package racingcar.domain;
 
-import java.util.ArrayList;
+import java.util.Collection;
 import java.util.List;
+import java.util.stream.Collectors;
 
-import static java.util.stream.Collectors.joining;
+import static java.util.Comparator.comparing;
 import static java.util.stream.Collectors.toList;
-import static racingcar.domain.RacingGameConstant.DELIMITER_COMMA;
 import static racingcar.domain.RacingGameConstant.NO_WINNER;
 
 public class Cars {
     private final List<Car> cars;
 
-    Cars(String[] carNames) {
-        List<Car> cars = new ArrayList<>();
-        for (int i = 0; i < carNames.length; i++) {
-            cars.add(new Car(carNames[i]));
-        }
-        this.cars = cars;
+    Cars(Collection<String> carNames) {
+        this.cars = carNames.stream()
+                .map(Car::new)
+                .collect(toList());
```

Comment:
```suggestion
    Cars(Collection<String> carNames) {
    public static Cars of(List<String> carNames) {
        return new Cars(carNames.stream()
                .map(Car::new)
                .collect(toList()));
    }
```

인자 타입이 겹쳐서 Collection으로 우회한 것으로 보이는데요. 정적 팩토리 메서드를 활용하는 것은 어떻게 생각하시나요?

> https://johngrib.github.io/wiki/static-factory-method-pattern/